bindings/python/setup.py

```diff
@@ -65,18 +66,22 @@ def __init__(self, name):
         super().__init__(name, sources=[])
 
 
-def escape(string):
-    return '"' + string.replace('\\', '\\\\') + '"'
+def b2_escape(value):
+    value = value.replace("\\", "\\\\")
+    value = value.replace('"', '\\"')
```

Comment:
did you try this with a path containing a double quote, or is this encoding an assumption? I'm not 100% sure this wouldn't need to be double-escaped, but probably not.

Reply:
I didn't test it yet. This is per https://boostorg.github.io/build/manual/develop/index.html#jam.language.lexical :

>B2 treats its input files as whitespace-separated tokens, with two exceptions: double quotes (") can enclose whitespace to embed it into a token, and everything between the matching curly braces (\{}) in the definition of a rule action is treated as a single string. A backslash (\) can escape a double quote, or any single whitespace character.

I originally read it such that `"` needs to be escaped as `\"` within a `"quoted string"`, but actually I guess it's ambiguous.

Although, it doesn't even say that `\` needs to be escaped as `\\`, though it occurs in their examples. Am I using the right documentation for B2? I found this documentation to be patchy and convoluted. I don't understand why the basic syntax is covered at the end of the manual, among other things

Reply:
If I stick a `using python ` line in my `user-config.jam` with paths like `"/home/foo/test \"test\"/python"` and enable debug output, it looks like b2 parses the config in the expected way.